t/quri.lisp

```diff
@@ -127,10 +127,14 @@
     (,(uri "https://example.org/#/?b") . "https://example.org/#/?b")
     (,(uri "about:blank") . "about:blank")))
 
-(subtest "merge-uris"
-  (loop for (test-uri . result-uri) in *merge-test-cases* do
-       (let ((merged-uri (merge-uris test-uri *base-uri*)))
-         (is (render-uri merged-uri) result-uri :test 'string=))))
+(loop for (test-uri . result-uri) in *merge-test-cases* do
+  (let ((merged-uri (merge-uris test-uri *base-uri*)))
+    (subtest "merge-uris"
+      (is (render-uri merged-uri) result-uri :test 'string=))
+    (subtest "merge-uris-type-check"
+      (unless (uri-scheme test-uri)
+        (is (symbol-name (type-of merged-uri))
+            (symbol-name (type-of *base-uri*)))))))
```

Comment:
I don't think `loop` should be moved outside, this way you'll get intertwined subtests for each entry.

Instead, copy the subtest, it's OK to copy a bit of trivial code.

Reply:
It works ok if a subset of the subtests fail. Also, that's the strategy adopted above on line 84.

If you're really set on this change, then let's change both loops.

Reply:
Oh, just saw that.  Yup, the tests are not very well written.

Oh well, that's fine then! :)